fireant/queries/builder/dataset_blender_query_builder.py

```diff
@@ -311,11 +339,30 @@ def sql(self):
         be produced.  This next line converts the list of rows of the table in the diagram to a list of columns. Each 
         set of queries in a column are then reduced to a single data blending sql query. 
         """
-        querysets_tx = list(
-            zip(*[dataset_query.sql for dataset_query in dataset_queries])
-        )
 
-        return [
-            _blend_query(self._dimensions, metrics, self.orders, field_maps, queryset)
-            for queryset in querysets_tx
-        ]
+        # TODO: what if some of the datasets end up doing more reference/total calculations? Make sure to prevent this.
+        per_dataset_queries_count = max(
+            [len(dataset_queries) for dataset_queries in datasets_queries]
+        )
+        query_sets = [[] for _ in range(per_dataset_queries_count)]
+
+        for dataset_queries in datasets_queries:
+            for i, dataset_query in enumerate(dataset_queries):
+                query_sets[i].append(dataset_query)
+
+        blended_queries = []
+        for queryset in query_sets:
+            # Remove trailing None's
+            while queryset[-1] is None:
```

Comment:
Out of interest - what is the scenario that some elements can be None? Are there cases when totals for example can be applied to one data set e.g. the primary and not a blended data set?

Reply:
So if you link multiple datasets, it would normally result in multiple subqueries right. But if you are linking 2 datasets, but you are only requesting a metric from the second one, this technically only results in a query for the 2nd dataset. The code however will still return None for that first dataset so that the blending code knows that the first dataset field map shouldn't be used. There are still more optimisations possible when it comes to leaving out certain subqueries btw, but I will leave that for later.

Reply:
Makes sense, thanks for sharing the info.